src/core/request.js

```diff
@@ -233,7 +233,7 @@ const request = async (ctx, uri, options) => {
       /* istanbul ignore else */
       if (opts.headers['transfer-encoding'] === undefined
         && opts.headers['content-length'] === undefined) {
-        opts.headers['content-length'] = String(opts.body.length);
+        opts.headers['content-length'] = Buffer.byteLength(opts.body, 'utf-8');
```

Comment:
@dominique-pfister @stefan-guggisberg what if the body is a buffer? isn't this wrong then?
wouldn't it be cleaner to calculate it above ?

AFAICS, a byte body still has a wrong content length (which might explain some errors we have in blob support)

Reply:
If the body **is** a Buffer, what method should we use instead? `Buffer.length`?

Reply:
yes. see https://github.com/adobe/helix-fetch/pull/155

Reply:
agreed, thanks for catching this!